### ISSUE TITLE
Metaqueries in Flux are inconvenient

### ISSUE THREAD

**Issue by wolffcm** [influxdata/flux#3618]
In InfluxQL we have commands like `SHOW TAG KEYS` and `SHOW FIELD KEYS` that could break out tag keys and field keys per measurement. They are intuitive and easy to use.

Flux is not so easy. For example, you can use `tagKeys(fn: (r) => true, ...)` to get all the tag keys in a bucket (with no indication of which tags are used by which measurement), or you can use `tagKeys(fn: (r) => r._measurement == "some_measurement")` to get tag keys for a particular measurement. But there is no easy way to get all the tag keys grouped by measurement for a bucket.

The situation for fields is similar.

I came up with this query that gets part of the way there:
```js
import "influxdata/influxdb/schema"
import "strings"

bucket = "telegraf"

schema.tagValues(bucket: bucket, tag: "_measurement")
  |> map(fn: (r) => {
    meas = r._value
    fields = schema.fieldKeys(bucket: bucket, predicate: (r) => r._measurement == meas) |> findColumn(column: "_value", fn: (key) => true)
    joinedFields = strings.joinStr(arr: fields, v: ", ")
    tags = schema.tagKeys(bucket: bucket, predicate: (r) => r._measurement == meas) 
      |> filter(fn: (r) => r._value != "_start" and r._value != "_stop" and r._value != "_measurement" and r._value != "_field")
      |> findColumn(column: "_value", fn: (key) => true)
    joinedTags = strings.joinStr(arr: tags, v: ", ")
    return {measurement: r._value, fields: joinedFields, tags: joinedTags}
  })
  |> group(columns: ["measurement"])
```

But this puts all of the tag keys and fields in the same row for each measurement. It would be better to have two queries (similar to InfluxQL) that create one row for each tag or field. I couldn't figure out how to do that in pure Flux though.

**Comment by wolffcm** [comment]
Note: unfortunately the above query produces a `panic` when I tried it in cloud 2.0:
https://github.com/influxdata/influxdb/issues/21186

**Comment by wolffcm** [comment]
Let's add this to the `experimental` package.

Once we have a `flatmap` implementation, we can use that to avoid having to do `joinStr`, which is problematic, since field names and tag keys can contain `,`. Flatmap issue: https://github.com/influxdata/flux/issues/2539

We need to fix the panic in influxdata/influxdb#21186 before this is really useful.

Because in Cloud this is essentially 2x round trips to storage per measurement, this might not be all that performant. In the future we can investigate creating new storage RPCs that would make this faster and or simplify logic on the query side.
